src/common/ThreadStart.cpp

```diff
@@ -309,13 +309,12 @@ ThreadId Thread::start(ThreadEntryPoint* routine, void* arg, int priority_arg, H
 	 * Advanced Windows by Richter pg. # 109. */
 
 	unsigned thread_id;
-	unsigned long real_handle =
-		_beginthreadex(NULL, 0, THREAD_ENTRYPOINT, THREAD_ARG, CREATE_SUSPENDED, &thread_id);
-	if (!real_handle)
+	HANDLE handle =
+		reinterpret_cast<HANDLE>(_beginthreadex(NULL, 0, THREAD_ENTRYPOINT, THREAD_ARG, CREATE_SUSPENDED, &thread_id));
+	if (!handle)
 	{
-		Firebird::system_call_failed::raise("_beginthreadex", GetLastError());
+		Firebird::system_call_failed::raise("_beginthreadex", errno);
 	}
```

Comment:
IIRC, system_call_failed / system_error exceptions expect Win32 error codes rather then CRT ones.

Reply:
Ok, I've reverted to GetLastError() with comment.